presto-native-execution/src/test/java/com/facebook/presto/spark/TestPrestoSparkNativeExecution.java

```diff
@@ -87,4 +93,30 @@ public void testNativeExecutionShuffleManager()
         assertEquals(0, shuffleId);
         assertEquals(shuffleManager.getNumOfPartitions(shuffleId), bypassMergeSortShuffleHandle.numMaps());
     }
+
+    // TODO: re-enable the test when presto native extends support beyond builtin functions.
+    @Ignore
+    /**
+     * Test native execution of a cpp function declared via json file.
+     */
+    public void testNativeExecutionWithJsonFileBasedFunction()
+    {
+        Session session = Session.builder(getNativeSession()).build();
+        PrestoSparkQueryRunner queryRunner = (PrestoSparkQueryRunner) getQueryRunner();
+        setupJsonFunctionNamespaceManager(queryRunner);
+        assertQuerySucceeds(session, "SELECT json.f3.eq(1,2) FROM nation");
```

Comment:
Can we enable the test, but change this to assertQueryFails ?

CC: @spershin

Reply:
I could but curious what's the benefit of that approach?

Reply:
There is no benefit in adding disabled tests. These tests do not run and do not provide any verification. Adding a test that runs a query and verifies a specific error message give some verification.

Reply:
The 'fail' approach is slightly better than 'disabled' in my opinion because:
1. 'disabled' would likely be forgotten.
2. 'fail' would start failing as soon as functionality added and will prompt the engineer to change that to a proper 'enabled' test.